handler/websocket.go

```diff
@@ -63,33 +63,42 @@ func connectWs(exec graphql.ExecutableSchema, w http.ResponseWriter, r *http.Req
 		cfg:    cfg,
 	}
 
-	if !conn.init() {
+	initPayload, ok := conn.init()
+	if !ok {
 		return
 	}
 
-	conn.run()
+	conn.run(initPayload)
 }
 
-func (c *wsConnection) init() bool {
+func (c *wsConnection) init() (initPayload graphql.InitPayload, ok bool) {
 	message := c.readOp()
 	if message == nil {
 		c.close(websocket.CloseProtocolError, "decoding error")
-		return false
+		return nil, false
 	}
 
+	initPayload = make(graphql.InitPayload)
+
 	switch message.Type {
 	case connectionInitMsg:
+		err := json.Unmarshal(message.Payload, &initPayload)
+		if err != nil {
+			// Treat an invalid payload as no payload
```

Comment:
maybe for an empty payload this is what you want, but squashing all errors isn't good.

Reply:
I could change that to only run this when there is a payload, and pass the error up.